query-engine/core/src/query_graph_builder/write/write_args_parser.rs

```diff
@@ -24,49 +26,13 @@ impl WriteArgsParser {
                 let field = model.fields().find_from_all(&k).unwrap();
 
                 match field {
-                    Field::Scalar(sf) if sf.is_list() => match v {
-                        ParsedInputValue::List(_) => {
-                            let set_value: PrismaValue = v.try_into()?;
-
-                            args.args.insert(sf, set_value);
-                        }
-                        ParsedInputValue::Map(map) => {
-                            let expr = extract_scalar_list_ops(map)?;
-
-                            args.args.insert(sf, expr)
-                        }
-                        _ => unreachable!(),
-                    },
+                    Field::Scalar(sf) if sf.is_list() => {
+                        let expr = parse_scalar_list(v)?;
 
+                        args.args.insert(sf, expr);
+                    }
```

Comment:
I couldn't split this whole bit into a reusable function in this end. The reason is `Field::Relation` which wouldn't return a `WriteExpression`